src/Common.OData.ApiExplorer/AspNet.OData/ClassProperty.cs

```diff
@@ -30,7 +30,18 @@ internal ClassProperty( IEnumerable<Assembly> assemblies, IEdmOperationParameter
             Contract.Requires( parameter != null );
 
             Name = parameter.Name;
-            type = parameter.Type.Definition.GetClrType( assemblies );
+
+            if ( parameter.Type.IsCollection() )
+            {
+                var collectionType = parameter.Type.AsCollection();
+                var elementType = collectionType.ElementType().Definition.GetClrType( assemblies );
```

Comment:
Don't we still have to handle possible substitution here if the element type is an _Entity Type_ or _Complex Type_? The element will be a type reference, but the registered reference _might_ have filtered members in this version, no?

Reply:
Yeah you're right.

I had to change the interface so a serviceprovider required for making the substitutioncontext is passed. I also added a new test for action parameters with types that should be substituted.